# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextProviderEGL.cpp
# Commit: 6841dc516087
# Full Hash: 6841dc516087239f3e48806a38356ad725c2c272
# Author: stransky <stransky@redhat.com>
# Date: 2022-09-21 21:43:38
# Regressor Bug: 1777664
# File Overlap Count: 1
# Description:
#   Bug 1645677 [Wayland] Update EGLSurface when wl_surface is deleted r=emilio,jgilbert
#   
#   When GtkWidget is hidden, underlying wl_surface is deleted. We need to also update EGLSurface of GtkWidget (GtkCompositorWidget)
#   as EGLSurface is directly linked to wl_surface:
#   
# ==============================================================================

diff -r 5ba3a56b3038 -r 6841dc516087 gfx/gl/GLContextProviderEGL.cpp
--- a/gfx/gl/GLContextProviderEGL.cpp	Wed Sep 21 10:32:44 2022 +0000
+++ b/gfx/gl/GLContextProviderEGL.cpp	Wed Sep 21 10:32:44 2022 +0000
@@ -346,6 +346,18 @@
   }
 
   MOZ_ASSERT(aCompositorWidget);
+#ifdef MOZ_WAYLAND
+  // RenderCompositorEGL does not like EGL_NO_SURFACE as it fallbacks
+  // to SW rendering or claims itself as paused.
+  // In case we're missing valid native window because aCompositorWidget hidden,
+  // just create a fallback EGLSurface.
+  // Actual EGLSurface will be created by widget code later when
+  // aCompositorWidget becomes visible.
+  if (widget::GdkIsWaylandDisplay() && aCompositorWidget->IsHidden()) {
+    mozilla::gfx::IntSize pbSize(16, 16);
+    return CreateWaylandBufferSurface(*egl, aConfig, pbSize);
+  }
+#endif
   EGLNativeWindowType window =
       GET_NATIVE_WINDOW_FROM_COMPOSITOR_WIDGET(aCompositorWidget);
   if (!window) {